package/MDAnalysis/core/selection.py

```diff
@@ -530,11 +530,23 @@ def __init__(self, parser, tokens):
 
     @return_empty_on_apply
     def apply(self, group):
-        mask = np.zeros(len(group), dtype=bool)
-        for val in self.values:
-            values = getattr(group, self.field)
-            mask |= [fnmatch.fnmatch(x, val) for x in values]
-        return group[mask].unique
+        # rather than work on group.names, cheat and look at the lookup table
+        nmattr = getattr(group.universe._topology, self.field)
+
+        matches = []  # list of passing indices
+        # iterate through set of known atom names, check which pass
+        for nm, ix in nmattr.namedict.items():
+            if any(fnmatch.fnmatch(nm, val) for val in self.values):
```

Comment:
I think we need to be using [fnmatchcase()](https://docs.python.org/3/library/fnmatch.html#fnmatch.fnmatchcase) so that it doesn't depend on your OS if you match "GLU" with "Glu" on Windows (where [os.path.normcase()](https://docs.python.org/3/library/os.path.html#os.path.normcase) is applied!)  but not on Linux.

Reply:
good catch, thanks